### PR TITLE
[ENHANCEMENT] Add timed jobs (Cron)

### PR DESCRIPTION
**Pull request by darknight2590**
```
-- Execute task 5:10, every day
function CronTask(d, h, m)
  print('Task done')
end

TriggerEvent('smallresources:server:cronrunAt', 5, 10, CronTask)

-- Execute task every monday at 18:30
function CronTask(d, h, m)
  if d == 1 then
    print('Task done')
  end
end

TriggerEvent('smallresources:server:cronrunAt', 18, 30, CronTask)
```

### CODE REVIEW

**Review by mjvanhaastert — COMMENTED** [review]
oh i like this functions

File: server/cron.lua [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	end
+	SetTimeout(60000, Tick)
+end
+
+LastTime = GetTime()
+Tick()
+
+AddEventHandler('smallresources:server:cronrunAt', function(h, m, cb)
+	RunAt(h, m, cb)
+end)
```
Comment (Z3rio) [inline server/cron.lua:42]: This is incredibly insecure, you can trigger this from any cheat and inject any code you want (which would run on the serverside).
If you want this feature then an export shall be used.
```suggestion
exports("cronrunAt", function(h, m, cb)
	RunAt(h, m, cb)
end)
```

**Review by Z3rio — CHANGES_REQUESTED** [review]
Incredibly good feature, although with an incredibly large exploit.

File: server/cron.lua (hunk 1 above)
Comment (Z3rio) [inline server/cron.lua:42]:
```suggestion
-- Variables
local Jobs = {}

-- Functions
function GetTime()
	local timestamp = os.time()
	local day = tonumber(os.date('*t', timestamp).wday)
	local hour = tonumber(os.date('%H', timestamp))
	local min = tonumber(os.date('%M', timestamp))

	return {day = day, hour = hour, min = min}
end

function CheckTimes(day, hour, min)
	for i,v in pairs(Jobs) do
		if v.hour == hour and v.min == min then
			v.cb(day, hour, min)
		end
	end
end

-- Exports
exports("CreateCronJob", function(hour, min, cb)
	if hour and type(hour) == "number" and min and type(min) == "number" and cb and (type(cb) == "function" or type(cb) == "table") then
		table.insert(Jobs, {
			min  = min,
			hour  = hour,
			cb = cb
		})
	else
		print("WARN: Invalid arguments for cronrunAt(hour, min, cb)")
	end
end)

-- Main Loop
Citizen.CreateThread(function()
	while true do
		local time = GetTime()
		CheckTimes(time.day, time.hour, time.min)

		Citizen.Wait(60 * 1000)
	end
end)
```
Refactor code + patch exploitation

**Review by Z3rio — CHANGES_REQUESTED** [review]

File: server/cron.lua [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	local hour = tonumber(os.date('%H', timestamp))
+	local min = tonumber(os.date('%M', timestamp))
+
+	return {day = day, hour = hour, min = min}
+end
+
+function CheckTimes(day, hour, min)
+	for i,v in pairs(Jobs) do
+		if v.hour == hour and v.min == min then
+			v.cb(day, hour, min)
```
Comment (Z3rio) [inline server/cron.lua:17]:
```suggestion
	for i = 1, #Jobs, 1 do
		local data = Jobs[i]
		if data.hour == hour and data.min == min then
			data.cb(day, hour, min)
```
Fix linting

**Review by Z3rio — CHANGES_REQUESTED** [review]
Fix linting

**Review by Z3rio — APPROVED** [review]
LGTM, good PR

**Pull request merged**